### ISSUE TITLE
Viewed Product Block is not shown after upgrading from 1.3.4 to 1.4.0

### ISSUE THREAD

**Issue by handoyo** [abantecart/abantecart-src#1718]
Hi core devs,

I found out that viewed products is not shown upon upgrading from 1.3.4 to 1.4.0.

I looked at the db and found out that:

![image](https://github.com/user-attachments/assets/8a2e292b-3b0b-4cb4-9fee-c9431565b0f0)

1. in ac_blocks table, the controller is viewed_products/viewed_products, not blocks/viewed_products.

- upgrade from 1.3.4 to 1.4.0

![image](https://github.com/user-attachments/assets/8db54d82-8491-4ab2-8647-76e6295400a6)

- 1.4.0
![image](https://github.com/user-attachments/assets/6fbdcd4c-ce79-44d2-9d12-3fa9ed64c9c0)

2. in ac_block_templates, the template doesn't have the "blocks/" prefix unlike the v1.4.0 one.

also i want to ask what is the correct template for the parent_block_id = 5?

thanks

**Comment by abolabo** [comment]
hmm.. i guess it's a name collision with extension templates. 
Template column must contains route from template directory. You can look into install. sql inside install directory. There are all routes with subfolder "blocks".

**Comment by handoyo** [comment]
I found this code in the abantecart_upgrade.php:

`$block_info['block_txt_id'] = 'viewed_products';
$block_info['controller'] = 'viewed_products/viewed_products';

$block_info['templates'] = [
    ['parent_block_txt_id' => 'header', 'template' => 'viewed_block_column_header.tpl'],
    ['parent_block_txt_id' => 'header_bottom', 'template' => 'viewed_block_column_header_bottom.tpl'],
    ['parent_block_txt_id' => 'content_top', 'template' => 'viewed_block_column_content_top.tpl'],
    ['parent_block_txt_id' => 'content_bottom', 'template' => 'viewed_block_column_content_bottom.tpl'],
    ['parent_block_txt_id' => 'footer', 'template' => 'viewed_block_column_footer.tpl'],
    ['parent_block_txt_id' => 'footer_top', 'template' => 'viewed_block_column_footer_top.tpl'],
    ['parent_block_txt_id' => 'column_left', 'template' => 'viewed_block_column_left.tpl'],
    ['parent_block_txt_id' => 'column_right', 'template' => 'viewed_block_column_right.tpl'],
];

$block_info['descriptions'] = [['language_name' => 'english', 'name' => 'Viewed Products']];

$layout = new ALayoutManager();
$layout->saveBlock($block_info);`

**Comment by abolabo** [comment]
@handoyo yes, my mistake. 
Fixed in our upgrade package tar.gz . 
Can you check it now with installation key abantecart_upgrade_140 ?

**Comment by handoyo** [comment]
I just test it.

I see that the controller in xx_blocks table is still 'viewed_products/viewed_products' not 'blocks/viewed_products'.

![image](https://github.com/user-attachments/assets/ffa2870d-eb34-4ff0-942a-9eceda01967a)

And what is the fix for existing admin that alreadt upgrade to 1.4.0?

**Comment by abolabo** [comment]
@handoyo just replace table prefix and run. Do not forget clear cache after that.

```
#fix or prior upgrade 1.3.4->1.4.0
UPDATE `ac_blocks`
SET controller='blocks/viewed_products'
WHERE controller='viewed_products/viewed_products' AND block_txt_id = 'viewed_products';

UPDATE `ac_block_templates`
SET template = CONCAT('blocks/',template)
WHERE SUBSTRING(template,1,13) = 'viewed_block_';
```

**Comment by handoyo** [comment]
Thanks for the fix @abolabo.
